File: scanner/corpora/java/positive/aes_cbc.java
import javax.crypto.Cipher;
import javax.crypto.KeyGenerator;
import javax.crypto.SecretKey;
import javax.crypto.spec.SecretKeySpec;
import javax.crypto.spec.IvParameterSpec;
import java.security.SecureRandom;

public class AesCbcExample {
    public static void main(String[] args) throws Exception {
        SecureRandom random = new SecureRandom();
        byte[] keyBytes = new byte[16];
        random.nextBytes(keyBytes);
        SecretKey key = new SecretKeySpec(keyBytes, "AES");

        byte[] iv = new byte[16];
        random.nextBytes(iv);
        IvParameterSpec ivSpec = new IvParameterSpec(iv);

        Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
        cipher.init(Cipher.ENCRYPT_MODE, key, ivSpec);
        String plaintext = "Secret CBC message";
        byte[] ciphertext = cipher.doFinal(plaintext.getBytes());

        Cipher cipherDec = Cipher.getInstance("AES/CBC/PKCS5Padding");
        cipherDec.init(Cipher.DECRYPT_MODE, key, ivSpec);
        byte[] decrypted = cipherDec.doFinal(ciphertext);
        System.out.println(new String(decrypted));
    }
}

File: scanner/corpora/python/positive/aes_cbc.py
"""AES-CBC encryption and decryption using PyCryptodome."""
from Crypto.Cipher import AES
from Crypto.Util.Padding import pad, unpad
from Crypto.Random import get_random_bytes

key = get_random_bytes(16)  # 128-bit key
iv = get_random_bytes(16)
cipher = AES.new(key, AES.MODE_CBC, iv)
padded_data = pad(b"Secret message for CBC mode", AES.block_size)
ciphertext = cipher.encrypt(padded_data)

# Decrypt
cipher_dec = AES.new(key, AES.MODE_CBC, iv)
decrypted_padded = cipher_dec.decrypt(ciphertext)
decrypted = unpad(decrypted_padded, AES.block_size)
print(decrypted)
